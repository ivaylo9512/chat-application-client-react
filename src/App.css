#root{
    overflow: hidden;
    position: relative;
    display: flex;
    flex-direction: column;
    height: 100vh;
}
#root > div,
#root > section{
    height: 100vh;
    overflow: hidden;
    position: relative;
    display: flex;
    flex-direction: column;
}
#root .type-buttons{
    flex-direction: row;
}
.content-container{
    border-top: 0.6vh solid rgb(24,27, 33);
}
.type-buttons button{
    flex-grow: 1;
}
.content{
    flex: 1;
    display: flex;
}
.chat-container{
    width: 100%;
    height: 100%;
    display: flex;
    flex-direction: column;
}
.chat,
.users,
.chats{
    width: 100%;
    flex: 1;
}