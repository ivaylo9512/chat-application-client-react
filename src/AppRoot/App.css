#root{
    overflow: hidden;
    position: relative;
    display: flex;
    flex-direction: column;
    height: 100%;
}
#root > div,
#root > section{
    height: 100%;
    min-height: -webkit-fill-available;
    overflow: hidden;
    position: relative;
    display: flex;
    flex-direction: column;
}
