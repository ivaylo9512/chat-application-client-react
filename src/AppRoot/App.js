import Login from 'components/Login/Login'
import Register from 'components/Register/Register'
import Logged from 'components/Logged/Logged'
import Logout from 'components/Logout/Logout'
import { Route, Switch, Redirect } from 'react-router-dom'
import { useSelector } from 'react-redux';
import { getIsAuth } from 'app/slices/authenticateSlice';

const App = () => {
    const isAuth = useSelector(getIsAuth);

    return (
        <>
            {isAuth
                ? <>
                    <Logged/> 
                    <Route path='/logout' render={() => <Logout />} />
                </> 
                : <Switch>
                    <Route path='/register' render={() => <Register />}/>
                    <Route data-testid='login' path='/login'render={() => <Login />}/>
                    <Redirect from='/' to='/login'/>
                </Switch>
            } 
        </>
    )
}

export default App;
