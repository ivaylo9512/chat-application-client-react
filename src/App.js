import './App.css';
import Login from './components/Login/Login'
import Register from './components/Register/Register'
import Logged from './components/Logged/Logged'
import { Route, Switch, Redirect, BrowserRouter as Router } from 'react-router-dom'
import ErrorBoundary from './utils/ErrorBoundary'
import { useDispatch, useSelector } from 'react-redux';
import { onLogout, getIsAuth } from './app/slices/authenticateSlice';

const App = () => {
    const isAuth = useSelector(getIsAuth);
    const dispatch = useDispatch();

    const logout = () => {
        localStorage.removeItem('Authorization');
        localStorage.removeItem('user');
        dispatch(onLogout());
    }

    return (
        <ErrorBoundary>
            <Router>
                {isAuth
                    ? <>
                        <Logged/> 
                        <Route path='/logout' render={logout}/>
                    </> 
                    : <Switch>
                        <Route path='/register' render={() => <Register />}/>
                        <Route path='/login'render={() => <Login />}/>
                        <Redirect from='/' to='/login'/>
                    </Switch>
                } 
            </Router>
        </ErrorBoundary>
    )
}

export default App;
