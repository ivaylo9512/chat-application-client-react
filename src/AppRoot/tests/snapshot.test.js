import App from 'AppRoot/App';
import { shallow } from 'enzyme';
import * as redux from 'react-redux';
describe('App snapshot tests', () => {
    beforeEach(() => {
        jest.spyOn(redux, 'useDispatch').mockReturnValue(jest.fn());
    })

    const createWrapper = (state) => {
        jest.spyOn(redux, 'useSelector').mockReturnValue(state);
        
        return shallow(<App />)
    }

    it('should render with logged user', () => {
        const wrapper = createWrapper(true);

        expect(wrapper).toMatchSnapshot();
    })

    it('should render without logged user', () => {
        const wrapper = createWrapper(false);

        expect(wrapper).toMatchSnapshot();
    })
})
