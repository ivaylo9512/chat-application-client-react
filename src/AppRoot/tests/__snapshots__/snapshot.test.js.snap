// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot tests should render with logged user 1`] = `
<Fragment>
  <Logged />
  <Route
    path="/logout"
    render={[Function]}
  />
</Fragment>
`;

exports[`App snapshot tests should render without logged user 1`] = `
<Fragment>
  <Switch>
    <Route
      path="/register"
      render={[Function]}
    />
    <Route
      data-testid="login"
      path="/login"
      render={[Function]}
    />
    <Redirect
      from="/"
      to="/login"
    />
  </Switch>
</Fragment>
`;
